### PR TITLE
server: refactor the GraphQL HTTP API handler to be generic

### PR DESCRIPTION
**Pull request by Naveenaidu**
### Description
Refactor the GraphQL HTTP API handler to be implementation agnostic. As a result we can add other implementations of the GraphQL HTTP handler.

### Related issue
https://github.com/hasura/graphql-engine-pro/issues/92

### Affected components

- [x] Server

### CODE REVIEW

File: server/src-lib/Hasura/Server/App.hs [hunk 1]
```diff
@@ -232,6 +232,12 @@ class Monad m => MonadConfigApiHandler m where
 -- instance (MonadIO m, UserAuthentication m, HttpLog m, Tracing.HasReporter m) => MonadConfigApiHandler (Tracing.TraceT m) where
 --   runConfigApiHandler = configApiGetHandler
 
+class Monad m => MonadGQLApiHandler m where
```
Comment (ecthiender) [inline server/src-lib/Hasura/Server/App.hs:235]: @Naveenaidu can you add some comments describing what is this for?

File: server/src-lib/Hasura/Server/App.hs [hunk 2]
```diff
@@ -539,6 +545,23 @@ configApiGetHandler serverCtx@ServerCtx{..} consoleAssetsDir =
...
+  ::  ( HasVersion
+      , MonadIO m
+      , UserAuthentication (Tracing.TraceT m)
+      , HttpLog m
+      , Tracing.HasReporter m
+      , E.MonadGQLExecutionCheck m
+      , MonadQueryLog m
+      , GH.MonadExecuteQuery m
+      , EQ.MonadQueryInstrumentation m
+      )
```
Comment (ecthiender) [inline server/src-lib/Hasura/Server/App.hs:558]: I think its a good idea to add these constraints to the `runGQLApiHandler` function as well.

File: server/src-lib/Hasura/Server/App.hs [hunk 3]
```diff
@@ -730,8 +755,7 @@ httpApp corsCfg serverCtx enableConsole consoleAssetsDir enableTelemetry = do
       Spock.post "v1alpha1/graphql" $ spockAction GH.encodeGQErr id $
         mkPostHandler $ mkAPIRespHandler $ v1Alpha1GQHandler E.QueryHasura
 
-      Spock.post "v1/graphql" $ spockAction GH.encodeGQErr allMod200 $
-        mkPostHandler $ mkAPIRespHandler v1GQHandler
+      runGQLApiHandler serverCtx
```
Comment (ecthiender) [inline server/src-lib/Hasura/Server/App.hs:758]: Maybe adding the following comment will help the future reader understand what's going on?

```suggestion
      -- See 'MonadGQLApiHandler'
      runGQLApiHandler serverCtx 
```

File: server/src-lib/Hasura/Server/App.hs [hunk 4]
```diff
@@ -232,6 +232,12 @@ class Monad m => MonadConfigApiHandler m where
 -- instance (MonadIO m, UserAuthentication m, HttpLog m, Tracing.HasReporter m) => MonadConfigApiHandler (Tracing.TraceT m) where
```
Comment (ecthiender) [inline server/src-lib/Hasura/Server/App.hs:232]: Also this is unrelated to your PR. But I think let's remove this commented out code.

**Review by ecthiender — CHANGES_REQUESTED** [review]
Overall LGTM. I only had some minor comments.

**Comment by ecthiender** [comment]
I think we should also add a default GET handler to execute graphql queries. It should expect `query`, `variables` and `operationName` as URL query string parameters. @0x777 @paf31 what do you think?

**Review by paf31 — APPROVED** [review]

**Comment by Stefmore02** [comment]
Closing this PR as changes reflected here have already been implemented.

**Pull request closed without merging**